src/libraries/Microsoft.Extensions.Hosting.Abstractions/src/HostingAbstractionsHostExtensions.cs

```diff
@@ -86,18 +86,18 @@ public static async Task RunAsync(this IHost host, CancellationToken token = def
         /// <returns>The <see cref="Task"/> that represents the asynchronous operation.</returns>
         public static async Task WaitForShutdownAsync(this IHost host, CancellationToken token = default)
         {
-            IHostApplicationLifetime applicationLifetime = host.Services.GetService<IHostApplicationLifetime>();
+            IHostApplicationLifetime applicationLifetime = host.Services.GetRequiredService<IHostApplicationLifetime>();
```

Comment:
I think this is an OK change, given that it will just NRE later if the service isn't there.

It also matches this code (which will run before this line is called:

https://github.com/dotnet/runtime/blob/385fc8832b1d4160abe13645054b65458b6a12f8/src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs#L256